src/workspaceViewer.ts

```diff
@@ -0,0 +1,150 @@
+import path = require('path');
+import { TreeDataProvider, EventEmitter, TreeItemCollapsibleState, TreeItem, Event, Uri, window } from 'vscode';
+import { runTextInTerm } from './rTerminal';
+import { globalenv, workingDir } from './session';
+import { config } from './util';
+
+interface WorkspaceAttr {
+    [key: string]: {
+        class: string[];
+        type: string;
+        length: number;
+        str: string;
+    }
+}
+export class WorkspaceDataProvider implements TreeDataProvider<WorkspaceItem> {
+	private _onDidChangeTreeData: EventEmitter<void> = new EventEmitter();
+	readonly onDidChangeTreeData: Event<void> = this._onDidChangeTreeData.event;
+
+	refresh(): void {
+		this.data = <WorkspaceAttr> globalenv;
+		this._onDidChangeTreeData.fire();
+	}
+
+	data: WorkspaceAttr;
+
+	getTreeItem(element: WorkspaceItem): TreeItem {
+		return element;
+	}
+
+	getChildren(): WorkspaceItem[] {
+		return this.getWorkspaceItems(this.data);
+	}
+
+	private getWorkspaceItems(data: WorkspaceAttr): WorkspaceItem[] {
+		const toItem = (
+			key: string,
+			rClass: string,
+			str: string,
+			type: string,
+			length: number
+		): WorkspaceItem => {
+			return new WorkspaceItem(
+				key,
+				rClass,
+				str,
+				type,
+				length,
+				TreeItemCollapsibleState.None
+			);
+		};
+
+		const items = data ? Object.keys(data).map((key) =>
+			toItem(
+				key,
+				data[key].class[0],
+				data[key].str,
+				data[key].type,
+				data[key].length
+			)) : [];
+
+		function sortItems(a: WorkspaceItem, b: WorkspaceItem) {
+			const priorityAttr: string[] = [
+				'list',
+				'environment'
+			]
+
+			if (priorityAttr.includes(a.contextValue) > priorityAttr.includes(b.contextValue)) {
+				return -1
+			} else if (priorityAttr.includes(b.contextValue) > priorityAttr.includes(a.contextValue)) {
+				return 1
+			} else {
+				return 0 || a.label.localeCompare(b.label);
+			}
+		}
+
+		return items.sort((a, b) => sortItems(a, b));
+	}
+}
+
+export class WorkspaceItem extends TreeItem {
+	constructor(
+		label: string,
+		rClass: string,
+		str: string,
+		type: string,
+		length: number,
+		collapsibleState: TreeItemCollapsibleState
+	) {
+		super(label, collapsibleState);
+		this.description = str;
+		this.tooltip = `${label} (${rClass}, length of ${length})`;
+		this.contextValue = type;
+	}
+}
+
+export function clearWorkspace(): void {
+	const removeHiddenItems: boolean = config().get('workspaceViewer.removeHiddenItems');
+	if (globalenv !== undefined) {
+		void window.showInformationMessage(
+			"Are you sure you want to clear the workspace? This cannot be reversed.",
+			"Confirm",
+			"Cancel"
+		).then(selection => {
+			if (selection === "Confirm") {
```

Comment:
Please use single quotes for strings instead of double quotes in the typescripts. Not sure why eslint does not complain. Will take a look later.

Reply:
Oops, sorry - still getting used to this! Will fix.

Reply:
@renkun-ken  I removed custom rules in #474.
It's my miss, and I just backed them now.

Reply:
@Ikuyadeu Thanks for the notice!